src/components/SensorMap/SensorMap.js

```diff
@@ -28,17 +38,35 @@ export const SensorMap = () => {
   if (loading) return <p>Loading</p>;
   if (!data) return <p>Not found</p>;
 
+
+
+  let finalData = data.MeanClimateMeasurements.filter(goodData => !data.sensorfaults.some(
+    faultySensor => goodData.loc_x === faultySensor.loc_x && goodData.loc_y === faultySensor.loc_y && goodData.floor === faultySensor.floor
+  ));
+
+  console.log(data.sensorfaults);
+
   return (
     <Container className='my-3 p-0'>
       <Row>
         <Col className='p-0'>
-          {data.MeanClimateMeasurements.map((sensor) => (
-            <Sensor
+          {finalData.map((sensor) => (
+            <Sensor key={[sensor.loc_x, sensor.loc_y, sensor.floor].join("-")}
+              {...sensor}
+              cellHeight={CELL_HEIGHT}
+              cellWidth={CELL_WIDTH}
+            />
+          ))}
+
+          {data.sensorfaults.map((sensor) => (
+            <Sensor key={[sensor.loc_x, sensor.loc_y, sensor.floor].join("-")}
               {...sensor}
               cellHeight={CELL_HEIGHT}
               cellWidth={CELL_WIDTH}
+              faulty = {true}
             />
           ))}
```

Comment:
Why do we have this duplication of rendering a sensor?

Reply:
@BoostedPenguin

Reply:
↑ true that. We could have better names though.